### PR TITLE
Frontend: Difficulty Selection/Home Page

### PR DESCRIPTION
**Pull request by hsiaojietng**
Frontend of the home page where users can select the difficulty of their interview coding question.

Homepage:
image

### CODE REVIEW

File: user-service/controller/user-controller.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ export async function signin(req, res) {
             }
             // Create JWT
             const token = jwt.sign(signedInUser, process.env.JWT_SECRET_KEY);
+            //Store JWT in local storage to check if user is logged in
+            localStorage.setItem('token', token);
```
[the remark this answers is not in the document]
Comment (hsiaojietng) [reply user-service/controller/user-controller.js:43]: I will create Github Issue for this function

[the remark this answers is not in the document]
Comment (DesmondTo) [reply user-service/controller/user-controller.js:43]: I think it's done by lixue already no worries

File: frontend/src/components/PrivateRoute.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Navigate, Outlet } from 'react-router-dom';
+
+function PrivateRoute() {
+    return !localStorage.getItem('token') ? <Navigate to='/signin' /> : <Outlet />;
```
Comment (DesmondTo) [inline frontend/src/components/PrivateRoute.js:4]: Hi, can you help us do a favor here thanks.

1. Create a new route in user-service that use the authenticateJwt function defined by lixue, can return a response telling you whether the token is valid. 

2. Then, in this file here you can send a GET request to user-service along with the cookie stored, using axios.

Refer to [this stackoverflow](https://stackoverflow.com/questions/43002444/make-axios-send-cookies-in-its-requests-automatically) to see how to do that.

3. The the res.success returned is true you navigate them to outlet, else to signin page

**Review by DesmondTo — CHANGES_REQUESTED** [review]

**Comment by hsiaojietng** [comment]
Original forked repo has some issues, thus I created another repo to fork the main project. This PR is linked to the previous repo that has issues, hence I will close this PR.

**Pull request closed without merging**